parser/src/ast/declarations.rs

```diff
@@ -0,0 +1,523 @@
+///! This module provides AST structures which represent declarations permitted at module scope.
+///!
+///! There are no expressions/statements permitted in the top-level of a module, only declarations.
+///! These declarations define named items which are used by functions/constraints during evaluation.
+///!
+///! Some declarations introduce identifiers at global scope, i.e. they are implicitly defined in all
+///! modules regardless of imports. These are the `trace_columns`, `public_inputs`, and `random_values`
+///! sections.
```

Comment:
Sure, that's simple to change - though it doesn't actually cause problems if those declarations are publically visible, and it definitely feels like a strange mix of visibilities from the root module, with only one of those key top-level declarations globally-visible while others aren't, especially because the `random_values` declaration doesn't have a fixed name. Library modules on their own are incomplete - they must be evaluated in the context of a root module to be meaningful, so referencing global declarations of any kind is likely to cause issues if they have different names.

That said, the fewer things globally visible the better.

Reply:
I agree that it's a strange mix of visibilities - I'd rather find an alternative way to deal with `random_values`, but we haven't come up with anything yet (we also haven't spent a lot of time on it).

Reply:
Probably the simplest would be to disallow naming the main random values declaration, i.e. changing the declaration syntax to either `random_values: [2]` or `random_values: [a, b]`; then making `$random` or `$rand` a predefined name to refer to the entire random_values array.